src/site/stages/build/drupal/api.js

```diff
@@ -167,20 +168,26 @@ function getDrupalClient(buildOptions, clientOptionsArg) {
       }
     },
 
-    getAllPages(onlyPublishedContent = true) {
+    async getAllPages(onlyPublishedContent = true) {
+      const allPagesQuery = await getQuery(queries.GET_ALL_PAGES);
```

Comment:
This looks fine. 

One recommendation I have would be to move the subQuery call here: 
```suggestion
      const { getSideNavsViaGraphQL } = require('../../metalsmith-drupal');
      ...
      const sideNavs = await getSideNavsViaGraphQL();
      const allPagesQuery = getQuery(queries.GET_ALL_PAGES, sideNavs);
```

This might remove the need to spread the async stuff around. The const `facilitySidebarQuery.compiledQuery()` could be replaced with something called `makeFacilitySidebarQuery(sidenavs)`. 

The idea is to push the async stuff as far up as possible and inject the result into the query generating functions- so there's only one part of the code that's dealing with "async" and then everything after is just synchronous.

Reply:
Thanks @rianfowler. Since this is a pretty large change, I'll log it as a tech debt ticket.

Reply:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/15816